lockbox-ios/Presenter/ItemDetailPresenter.swift

```diff
@@ -75,79 +78,93 @@ class ItemDetailPresenter {
+                            }
 
-                    let message = String(format: Constant.string.fieldNameCopied, fieldName)
-                    self.view?.displayTemporaryAlert(message, timeout: Constant.number.displayStatusAlertLength)
-                })
-                .disposed(by: self.disposeBag)
+                            let message = String(format: Constant.string.fieldNameCopied, fieldName)
+                            self.view?.displayTemporaryAlert(message, timeout: Constant.number.displayStatusAlertLength)
+                        })
+                        .disposed(by: self.disposeBag)
+        })
+        .disposed(by: self.disposeBag)
 
         self.view?.learnHowToEditTapped
-                .subscribe { _ in
-                    guard let itemID = self.view?.itemId else {
-                        return
-                    }
-
-                    self.dispatcher.dispatch(action:
+            .subscribe { _ in
+                self.itemDetailStore.itemDetailId
+                    .take(1)
+                    .subscribe(onNext: { itemId in
+                        self.dispatcher.dispatch(action:
                             ExternalWebsiteRouteAction(
-                                    urlString: Constant.app.editExistingEntriesFAQ,
-                                    title: Constant.string.faq,
-                                    returnRoute: MainRouteAction.detail(itemId: itemID))
-                    )
-                }
-                .disposed(by: self.disposeBag)
-
-        self.view?.enableBackButton(enabled: !tabletHelper.shouldDisplaySidebar)
+                                urlString: Constant.app.editExistingEntriesFAQ,
+                                title: Constant.string.faq,
+                                returnRoute: MainRouteAction.detail(itemId: itemId))
+                        )
+                    }).disposed(by: self.disposeBag)
+            }
+            .disposed(by: self.disposeBag)
+
+        self.sizeClassStore.shouldDisplaySidebar
+            .subscribe(onNext: { (enableSidebar) in
+                self.view?.enableBackButton(enabled: !enableSidebar)
+            })
+            .disposed(by: self.disposeBag)
     }
 
-    func dndStarted(itemId: String, value: String?) {
-        self.dispatcher.dispatch(action: DataStoreAction.touch(id: itemId))
-        self.dataStore.get(itemId)
+    func dndStarted(value: String?) {
+        self.itemDetailStore.itemDetailId
             .take(1)
-            .subscribe(onNext: { item in
-                let copyAction = ItemDetailPresenter.getCopyActionFor(item, value: value, actionType: .dnd)
-                self.dispatcher.dispatch(action: copyAction)
-            }).disposed(by: self.disposeBag)
+            .subscribe(onNext: { itemId in
+                self.dispatcher.dispatch(action: DataStoreAction.touch(id: itemId))
+
+                self.dataStore.get(itemId)
```

Comment:
more subscriptions in subscriptions! would be nice to lift this out like:

```swift
self.itemDetailStore.itemDetailId
    .take(1)
    .switchMap { self.dataStore.get($0) }
    .take(1)
    .map { item -> [Action] in 
        // touch & copy actions generated
    }
    .subscribe { // dispatch actions }
    .disposed(by: self.disposeBag)
```

aside: Now that I'm looking at all these places where I'm suggesting an `Observable<[Action]>` as the final before the subscription, I wonder if there's a way to map an `Observable<[T]>` into an `Observable<T>` with all the constituent array elements.... not necessary, but would be another good tidiness improvement to avoid the `actions.forEach` call I suggested above and allow you to just put `@selector(dispatcher::dispatch)` (in proper swifty syntax :p )